### ISSUE TITLE
Number of mo elements for vartriangleleft vs urcorner 

### ISSUE THREAD

**Issue by physikerwelt** [mathjax/MathJax#1212]
On https://phabricator.wikimedia.org/T105318 @fred-wang wrote:

```
I don't whether it's a bug or a feature but on https://en.wikipedia.org/wiki/Help:Displaying_a_formula

I see that

    "\ulcorner \urcorner \llcorner \lrcorner" generates 4 characters, each one in its own <mo> element while

    while "\vartriangleleft \ntriangleleft \vartriangleright \ntriangleright" puts the 4 characters in the same <mo> element.

I'm not sure this situation happens in "real" math, though.
```
Can someone explan why this is the case?

**Comment by pkra** [comment]
tl;dr This is expected behavior due to MathJax trying hard to both be faithful to TeX and produce good MathML.

The second string consists of relations, the first does not. In TeX, adjacent relations get no space between them. This makes a lot of sense in real-world content because consecutive relations such as `:=` form a single unit both typographically and semantically.

To convert this to a single `<mo>` is sensible both in terms of typography and semantics suggested by TeX. Semantically, a single relation should be a single `<mo>`. Typographically, two consecutive `<mo>`'s would get spacing in between them, so they would not render the same way as the TeX input indicates. 

Of course, if you disagree with the semantic interpretation, you could manually adjust spacing, e.g., `<mo rspace="0">:</mo><mo lspace="0">=</mo>`, but that does not seem very good MathML and so MathJax does not produce this output.
